### PR TITLE
Wire Https Port to Server Games

### PR DESCRIPTION
**Pull request by DanVanAtta**
- Use LobbyServerProperties object to pass lobby properties, this results in simplifications
where we no longer need to get lobby properties from LobbyClient

- Pass HTTPs port to bot via args.

Net result is HTTPS port should become available to hosted games.

 

## Functional Changes

[ ] New map or map update
[ ] New Feature
[x] Feature update or enhancement
[ ] Feature Removal
[x] Code Cleanup or refactor
[ ] Configuration Change
[ ] Bug fix:  
[ ] Other:   

## Testing

[x] Covered by existing automated tests
[ ] Covered by newly added automated tests
[ ] Manually tested
[ ] No testing done

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyGamePanel.java [hunk 1]
```diff
@@ -155,7 +161,10 @@ private void mouseOnGamesList(final MouseEvent e) {
 
     final JPopupMenu menu = new JPopupMenu();
 
-    getUserGamesListContextActions().forEach(menu::add);
+    Arrays.asList(
+            SwingAction.of("Join Game", this::joinGame),
+            SwingAction.of("Host Game", () -> hostGame(lobbyServerProperties)))
+        .forEach(menu::add);
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyGamePanel.java]: I don't think having a list makes this more readable at this point.

File: .travis/run_smoke_tests [hunk 2]
```diff
@@ -74,6 +74,7 @@ function startBot() {
     -Dtriplea.name=Bot01Testing \
     -Dtriplea.lobby.host=localhost \
     -Dtriplea.lobby.port=3304 \
+    -Dtriplea.lobby.https.port=8080 \
```
Comment (RoiEXLab) [inline .travis/run_smoke_tests]: Typically HTTPS is 443, whereas 8080 is typically used as alternative port for HTTP (80) which might be confusing.

Anyways what I'm trying to say is that IMO this suggests that the transport (i.e. TLS) is an essential part for the communication to work, so if `https` is just supposed to represent the HTTP based lobby system, I'd actually vote to drop the s.

Comment (DanVanAtta) [reply .travis/run_smoke_tests]: I don't think I'm really thrilled that we call it 'https' or 'http' at all, we're bubbling up technology details.

We do call it 'https' in other places, notably client settings, I think we shot ourselves in the foot already. Of course, until we get some local certs installed and start using https locally, the port must be 8080. The intent is that we will get it switched to be https, having https not be available locally would be a bad move.

File: game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyGamePanel.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyGamePanel.java]: Personally I think that says something about the helper method: ` getUserGamesListContextActions()`.
Having the values be one per line IMO is a readability win.

To get parity with before, we could extract the list to an intermediate variable, but I think that would not be much cleaner in the end and highlights the suspect extracted method (which was excessively indirect)

**Pull request merged**